precompile/allowlist/role.go

```diff
@@ -5,22 +5,29 @@ package allowlist
 
 import "github.com/ethereum/go-ethereum/common"
 
+// 1. NoRole - this is equivalent to common.Hash{} and deletes the key from the DB when set
+// 2. EnabledRole - allowed to call the precompile
+// 3. Admin - allowed to both modify the allowlist and call the precompile
+// 4. Manager - allowed to add and remove only enabled address (only after DUpgrade), without being able to call the precompile.
+var (
+	// NoRole - this is equivalent to common.Hash{} and deletes the key from the DB when set.
+	NoRole = Role(common.BigToHash(common.Big0))
+	// EnabledRole - allowed to call the precompile.
+	EnabledRole = Role(common.BigToHash(common.Big1))
+	// Admin - allowed to both modify the allowlist and call the precompile.
+	AdminRole = Role(common.BigToHash(common.Big2))
+	// Manager - allowed to add and remove only enabled address, without being able to call the precompile (i.e not enabled). Activated only after DUpgrade.
+	// We don't specifically give enabled role to the manager, because we want to keep the backward compatibility with IsEnabled() function.
+	// TODO: we can give enabled role to the manager. For that we need to remove the IsEnabled(), so we can break other precompiles and force them to adopt.
```

Comment:
👁️

Reply:
I guess we have to give enabled to the manager because `TxAllowList` checks if the role is enabled to accept a tx. Meaning that even if manager role can alter someone else's enabled role, they won't be able to issue the tx (because their tx will be rejected by `preCheck` in [state_transition](https://github.com/ava-labs/subnet-evm/blob/master/core/state_transition.go#L350))